### PR TITLE
Delay proxy port release

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -431,7 +428,19 @@ func (p *Proxy) RemoveRedirect(id string, completions policy.CompletionContainer
...
+	// to be safe to listen on the port again
+	go func() {
+		time.Sleep(portReleaseDelay)
+
+		p.mutex.Lock()
+		delete(p.allocatedPorts, toPort)
+		p.mutex.Unlock()
+
+		log.WithField(fieldProxyRedirectID, id).
+			Debugf("Delayed release of proxy port 5d", toPort)
```
Comment (ianvernon) [inline pkg/proxy/proxy.go]: 5d --> %d

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: pkg/proxy/proxy.go (hunk 1 above)
Comment (tgraf) [reply pkg/proxy/proxy.go]: fixed

**Comment by tgraf** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: pkg/proxy/proxy.go [hunk 2]
```diff
@@ -431,7 +444,19 @@ func (p *Proxy) RemoveRedirect(id string, completions policy.CompletionContainer
 	r.Close(completions)
 
 	delete(p.redirects, id)
-	delete(p.allocatedPorts, toPort)
+
+	// delay the release and reuse of the port number so it is guaranteed
+	// to be safe to listen on the port again
+	go func() {
+		time.Sleep(portReleaseDelay)
```
Comment (eloycoto) [inline pkg/proxy/proxy.go:451]: Minor nitpick, here I would use `time.After` instead of the sleep.

**Pull request merged**